subsys/net/ip/net_context.c

```diff
@@ -591,15 +591,20 @@ int net_context_bind(struct net_context *context, const struct sockaddr *addr,
 		net_sin6_ptr(&context->local)->sin6_family = AF_INET6;
 		net_sin6_ptr(&context->local)->sin6_addr = ptr;
 		if (addr6->sin6_port) {
-			ret = check_used_port(AF_INET6, addr6->sin6_port,
+			ret = check_used_port(context->proto, addr6->sin6_port,
 					      addr);
-			if (!ret) {
-				net_sin6_ptr(&context->local)->sin6_port =
-					addr6->sin6_port;
-			} else {
+			if (ret != 0
+#if defined(CONFIG_NET_CONTEXT_REUSEADDR)
+			    && context->options.reuseaddr == 0
+#endif
+			    ) {
```

Comment:
This is quite hard to read with the ifdef inside if condition - IMO it'd make sense to introduce a helper function which provides the option value (`bool is_reuseaddr_set(struct net_context *context)` or so), that would simply return false if the option was disabled in KConfig.

Reply:
Done in the new commit